FrameworkSystem/Client/NotificationClient.py

```diff
@@ -48,131 +40,70 @@ def sendMail( self, addresses, subject, body,
   # ALARMS
   ###########################################################################
 
-  def newAlarm( self, subject, status, notifications, assignee, body, priority, alarmKey = "" ):
-    rpcClient = self.__getRPCClient()
-    if not isinstance( notifications, (list, tuple) ):
-      return S_ERROR( "Notifications parameter has to be a list or a tuple with a combination of [ 'Web', 'Mail', 'SMS' ]" )
-    alarmDef = { 'subject' : subject, 'status' : status,
-                 'notifications' : notifications, 'assignee' : assignee,
-                 'priority' : priority, 'body' : body }
+  def newAlarm(self, subject, status, notifications, assignee, body, priority, alarmKey=""):
+    if not isinstance(notifications, (list, tuple)):
+      return S_ERROR(
+          "Notifications parameter has to be a list or a tuple with a combination of [ 'Web', 'Mail', 'SMS' ]")
+    alarmDef = {'subject': subject, 'status': status,
+                'notifications': notifications, 'assignee': assignee,
+                'priority': priority, 'body': body}
     if alarmKey:
-      alarmDef[ 'alarmKey' ] = alarmKey
-    return rpcClient.newAlarm( alarmDef )
+      alarmDef['alarmKey'] = alarmKey
+    return self._getRPC().newAlarm(alarmDef)
 
-  def updateAlarm( self, id = -1, alarmKey = "", comment = False, modDict = {} ):
-    rpcClient = self.__getRPCClient()
+  def updateAlarm(self, id=-1, alarmKey="", comment=False, modDict={}):
     if id == -1 and not alarmKey:
-      return S_ERROR( "Need either alarm id or key to update an alarm!" )
-    updateReq = { 'comment' : comment, 'modifications' : modDict }
+      return S_ERROR("Need either alarm id or key to update an alarm!")
+    updateReq = {'comment': comment, 'modifications': modDict}
     if id != -1:
-      updateReq[ 'id' ] = id
+      updateReq['id'] = id
     if alarmKey:
-      updateReq[ 'alarmKey' ] = alarmKey
-    return rpcClient.updateAlarm( updateReq )
-
-  def getAlarms( self, selectDict, sortList, startItem, maxItems ):
-    rpcClient = self.__getRPCClient()
-    return rpcClient.getAlarms( selectDict, sortList, startItem, maxItems )
-
-  def getAlarmInfo( self, alarmId ):
-    rpcClient = self.__getRPCClient()
-    return rpcClient.getAlarmInfo( alarmId )
-
-  def deleteAlarmsById( self, alarmIdList ):
-    rpcClient = self.__getRPCClient()
-    return rpcClient.deleteAlarmsByAlarmId( alarmIdList )
-
-  def deleteAlarmsByKey( self, alarmKeyList ):
-    rpcClient = self.__getRPCClient()
-    return rpcClient.deleteAlarmsByAlarmKey( alarmKeyList )
-
-  ###########################################################################
-  # MANANGE ASSIGNEE GROUPS
-  ###########################################################################
-
-  def setAssigneeGroup( self, groupName, userList ):
-    rpcClient = self.__getRPCClient()
-    return rpcClient.setAssigneeGroup( groupName, userList )
-
-  def getUsersInAssigneeGroup( self, groupName ):
```

Comment:
You drop these methods because there are implicitly accessible through the generic client anyway ? If so, then we rather move in a different direction to specify explicitly all methods in a specific client in order to be able to specify individual timeout, for example, and also to make those methods visible at the iPython prompt.

Reply:
Yes I removed them because I am using `Client` as base class. 
The have calls visible through iPython we should improve the `Client` class instead: some time ago it was tried already https://github.com/DIRACGrid/DIRAC/blob/integration/Core/Base/Client.py#L34